### PR TITLE
[MRG] scale_source_space(): add support for vol source spaces

### PR DESCRIPTION
**Pull request by christianbrodbeck**
Resurrecting #1205...

### CODE REVIEW

**Comment by agramfort** [comment]
you seem to ahve some legit travis failures

**Comment by christianbrodbeck** [comment]
Ah, so the `fsaverage` in the testing environment does not have the MRI images... And it looks like when creating a source space without MRI image then it becomes a discrete source space, and after saving and reloading the `src_mri_t` entry is gone... Using the sample data (which has MRI images) instead makes the test go way longer, so I should probably stick with the discrete source space and just remove the test of `src_mri_t`?

**Comment by larsoner** [comment]
> Using the sample data (which has MRI images) instead makes the test go way longer

This is probably a function of your spacing. Change it to 2cm and it should be fast. If it's not, we should look at which step is slow.

(Another alternative: you can hack it by copying the `fsaverage` dir to a temp dir and copy in `sample`'s `T1.mgz`. It won't align properly but it hopefully won't matter for testing purposes. But hopefully we can get `sample` to work instead.)

**Comment by agramfort** [comment]
still WIP ? or MRG now?

**Comment by christianbrodbeck** [comment]
> This is probably a function of your spacing. Change it to 2cm and it should be fast. If it's not, we should look at which step is slow.

I am already using `pos=50`. The difference in in timing seems to come from just using the sample dataset, not form enabling the MRI test. I think it's because `create_default_subject()` copies all files in the subject's directory, and the fsaverage in the `sample` dataset has a lot more files, so this will in turn cause `scale_mri` to scale a lot more files. Trying by copying just the MRI file from sample data now...

**Comment by christianbrodbeck** [comment]
This fixed the tests, so I will set this to MRG

File: mne/tests/test_coreg.py [hunk 1]
```diff
@@ -1,7 +1,8 @@
 from glob import glob
 import os
+from shutil import copyfile
 
-from nose.tools import assert_equal, assert_raises, assert_true
+from nose.tools import assert_equal, assert_raises, assert_true, assert_is_not
```
Comment (larsoner) [inline mne/tests/test_coreg.py]: FYI we have gone to pytest so you can just do `assert x is not y` in tests now

**Comment by christianbrodbeck** [comment]
I think it added about a second on my Mac mini.

File: mne/tests/test_coreg.py (hunk 1 above)
Comment (christianbrodbeck) [reply mne/tests/test_coreg.py]: Do you want me to update the tests in this PR?

Comment (larsoner) [reply mne/tests/test_coreg.py]: Only if you are bored :)

**Comment by agramfort** [comment]
@larsoner merge if you're happy too

**Comment by larsoner** [comment]
@christianbrodbeck I don't get `nan` when doing this:
```
np.linalg.norm(np.array([-0.0005, -0.0005,  0.0005], dtype=np.float32))
```
So I am a bit confused by that commit message. But we can merge in any case.

**Comment by christianbrodbeck** [comment]
It might be the specific version/platform/distribution combination? I figured using `scipy`'s norm wouldn't hurt.

```
In [2]: import numpy as np

In [3]: np.__version__
Out[3]: '1.11.3'

In [4]: np.linalg.norm(np.array([-0.0005, -0.0005,  0.0005], dtype=np.float32))
//anaconda/envs/mayavi45/lib/python2.7/site-packages/numpy/linalg/linalg.py:2130: RuntimeWarning: invalid value encountered in sqrt
  ret = sqrt(sqnorm)
Out[4]: nan
```

**Pull request merged**